Buddies.API/Controllers/LeaderboardController.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Buddies.API.Entities;
+using Buddies.API.Database;
+using Microsoft.AspNetCore.Identity;
+using Buddies.API.IO;
+using Buddies.API.Services;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Authorization;
+using System.Text;
+
+namespace Buddies.API.Controllers
+{
+    [Route("api/v{version:apiVersion}/[controller]")]
+    [ApiController]
+    public class LeaderboardController : ControllerBase
+    {
+        private readonly ApiContext _context;
+        /// <summary>
+        /// Initializes a new NotificationController.
+        /// </summary>
+        /// <param name="userManager">UserManager from ASP.NET Core Identity.</param>
+        /// /// <param name="context">database context from Buddies.API.Database.</param>
+        public LeaderboardController(ApiContext context, UserManager<User> userManager)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        /// API route GET /api/v1/postings/{filters}/... for fetching all projects
+        /// that pass the filters.
+        /// </summary>
+        [HttpGet("{page}/{results}")]
+        public async Task<ActionResult> GetLeaderboard(int page, float results)
+        {
+            var userList = await _context.Users
+                .Include(u => u.Profile).ToListAsync();
```

Comment:
Ideally, you would want to do pagination here to avoid having to query for every user in the DB.

Reply:
yup!